utils/firestore.js

```diff
@@ -109,6 +109,25 @@ const updateResponse = async (data, uid) => {
         const response = await db.collection('participants').where('state.uid', '==', uid).get();
         if(response.size === 1) {
             for(let doc of response.docs){
+                if(data[399159511] && data[996038075]) { // if user profile F+L name is updated
+                    const previousQuery = doc.data().query
+                    const firstName = data[399159511].trim().toLowerCase()
```

Comment:
I would suggest moving this logic to `submit()` instead of inside this function... Also please condense this code, there are multiple statements made over and over. Could be much simpler.

Reply:
My bad just optimized. I need access to users`doc.data()` to access previous query that's the reason I wrote the code here instead of `submit()`.